src/cast_core/src/CommitLatency.cpp

```diff
@@ -0,0 +1,227 @@
+          useSession{phaseContext.get<bool, false>("Session").value_or(false)},
+          useTransaction{phaseContext.get<bool, false>("Transaction").value_or(false)},
+          repeat{phaseContext.get<IntegerSpec>("Repeat")},
+          threads{phaseContext.get<IntegerSpec>("Threads")},
+          amountDistribution{-100, 100},
+          options{ExecutionStrategy::getOptionsFrom(phaseContext, "ExecutionsStrategy")} {
+              // write_concern
+              if( phaseContext.get<bool, false>("WriteConcernMajority").value_or(false) ) {
+                  wc.majority(std::chrono::milliseconds{0});
+              }
+              else {
+                  wc.nodes( (std::int32_t) phaseContext.get<IntegerSpec, false>("WriteConcern").value_or(1) );
+              }
+              if( phaseContext.get<bool, false>("WriteConcernJournal").value_or(false) ) {
+                  wc.journal(true);
+              }
+              if( ! phaseContext.get<bool, false>("WriteConcernJournal").value_or(true) ) {
+                  // On some mongod versions it makes a difference whether journal is unset vs explicitly set to false
+                  wc.journal(false);
+              }
+              if ( useTransaction ) {
+                  optionsTransaction.write_concern(wc);
+              }
+              else {
+                  optionsUpdate.write_concern(wc);
+              }
+
+              // read_concern
+              rc.acknowledge_string(phaseContext.get<std::string>("ReadConcern"));
+              // Ugh... read_concern cannot be set for operations individually, only through client, database, or collection.
+              collection.read_concern(rc);
+              // Despite what documentation said, it can be set for transactions. Which is nice, as transactions will ignore the
+              // options set on the collection.
+              // TODO: File DOCS ticket to fix the self-contradicting mongocxx documentation.
+              optionsTransaction.read_concern(rc);
+
+              // read_preference
+              if ( rp_string == "PRIMARY" ) {
+                  // TODO: This actually sends primaryPreferred to server. -> File bug in Jira.
+                  // This is benign for this test, but potentially bad for real apps!
+                  // Note that k_secondary works correctly.
+                  rp.mode(mongocxx::read_preference::read_mode::k_primary);
+              }
+              else if ( rp_string == "SECONDARY" ) {
+                  rp.mode(mongocxx::read_preference::read_mode::k_secondary);
+              }
+              else {
+                  BOOST_THROW_EXCEPTION(InvalidConfigurationException("ReadPreference must be PRIMARY or SECONDARY."));
+              }
+              if ( useTransaction ) {
+                  optionsTransaction.read_preference(rp);
+              }
+              else {
+                  optionsFind.read_preference(rp);
```

Comment:
Since you're only setting read and write preferences on find, update and aggregate, I think you can just set them directly on the collection. This way you don't need to store `optionsFind`, `optionsUpdate`, and `optionsAggregate`

Reply:
Noted. My preference is still to add a comment here to indicate your concern since you're never setting `optionsTransaction` and on the collection at the same time.

No strong feelings though. Feel free to keep it as is.

Reply:
Comments have been updated and reference jira tickets too.